diff --git a/pywal/__main__.py b/pywal/__main__.py
index 91ead11..6141f7d 100644
--- a/pywal/__main__.py
+++ b/pywal/__main__.py
@@ -200,12 +200,12 @@ def parse_args(parser):
     if args.p:
         theme.save(colors_plain, args.p, args.l)
 
-    sequences.send(colors_plain, to_send=not args.s, vte_fix=args.vte)
-
-    if sys.stdout.isatty():
-        colors.palette()
-
-    export.every(colors_plain)
+    # sequences.send(colors_plain, to_send=not args.s, vte_fix=args.vte)
+    
+    if sys.stdout.isatty():
+        colors.palette()
+    
+    export.every(colors_plain)
 
     if not args.e:
         reload.env(tty_reload=not args.t)
