diff --git i/src/lread.c w/src/lread.c
index 3eef9f0ac9c..ea626dfae8d 100644
--- i/src/lread.c
+++ w/src/lread.c
@@ -1519,11 +1519,9 @@ DEFUN ("load", Fload, Sload, 1, 5, 0,
     Vloads_in_progress = Fcons (found, Vloads_in_progress);
   }
 
-  /* All loads are by default dynamic, unless the file itself specifies
-     otherwise using a file-variable in the first line.  This is bound here
-     so that it takes effect whether or not we use
-     Vload_source_file_function.  */
-  specbind (Qlexical_binding, Qnil);
+  /* All loads are by default lexical. This is bound here so that it
+     takes effect whether or not we use Vload_source_file_function.  */
+  specbind (Qlexical_binding, Qt);
 
   Lisp_Object found_eff =
     is_native_elisp
@@ -2418,12 +2416,11 @@ readevalloop (Lisp_Object readcharfun,
   record_unwind_protect_int (readevalloop_1, load_convert_to_unibyte);
   load_convert_to_unibyte = !NILP (unibyte);
 
-  /* If lexical binding is active (either because it was specified in
-     the file's header, or via a buffer-local variable), create an empty
+  /* If lexical binding is active (which is our default now), create an empty
      lexical environment, otherwise, turn off lexical binding.  */
   lex_bound = find_symbol_value (Qlexical_binding);
   specbind (Qinternal_interpreter_environment,
-	    (NILP (lex_bound) || BASE_EQ (lex_bound, Qunbound)
+	    (BASE_EQ (lex_bound, Qunbound)
 	     ? Qnil : list1 (Qt)));
   specbind (Qmacroexp__dynvars, Vmacroexp__dynvars);
 
@@ -2611,10 +2608,7 @@ DEFUN ("eval-buffer", Feval_buffer, Seval_buffer, 0, 5, "",
   record_unwind_protect_excursion ();
   BUF_TEMP_SET_PT (XBUFFER (buf), BUF_BEGV (XBUFFER (buf)));
   lexical_cookie_t lexc = lisp_file_lexical_cookie (buf);
-  specbind (Qlexical_binding,
-	    lexc == Cookie_Lex ? Qt
-	    : lexc == Cookie_Dyn ? Qnil
-	    : Fdefault_toplevel_value (Qlexical_binding));
+  specbind (Qlexical_binding, Qt);
   BUF_TEMP_SET_PT (XBUFFER (buf), BUF_BEGV (XBUFFER (buf)));
   readevalloop (buf, 0, filename,
 		!NILP (printflag), unibyte, Qnil, Qnil, Qnil);
@@ -6030,7 +6024,7 @@ syms_of_lread (void)
 This variable is automatically set from the file variables of an
 interpreted Lisp file read using `load'.  Unlike other file local
 variables, this must be set in the first line of a file.  */);
-  Vlexical_binding = Qnil;
+  Vlexical_binding = Qt;
   Fmake_variable_buffer_local (Qlexical_binding);
 
   DEFVAR_LISP ("eval-buffer-list", Veval_buffer_list,
